Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -12270,8 +12283,127 @@ def get_server_config():
+            :param profiler: The Profile object to use for profiling.
+            """
+
+            def dump_result():
+                """
+                Helper function to dump the profiling results to a file and return the file.
+                """
+                import tempfile
+                # Create a temporary file to store profiling data
+                with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+                    profiler.dump_stats(temp_file.name)
+                    temp_file_path = temp_file.name
+
+                # Read the profiling data from the temporary file
+                with open(temp_file_path, 'rb') as f:
+                    profiling_results = f.read()
+
+                # Delete the temporary file
+                os.remove(temp_file_path)
+
+                # Use Demisto's fileResult to create a file from the profiling stats
+                demisto.results(fileResult('profiling_stats.prof', profiling_results))
+
+            # 5 minutes in nanoseconds
+            default_timeout = 60 * 5 * 1e9
+
+            # The system timeout configuration.
+            timeout_nanoseconds = demisto.callingContext["context"].get("TimeoutDuration") or default_timeout
+            timeout_seconds = timeout_nanoseconds / 1e9
+            failed_on_timeout = False
+            # Set a timer to dump the file 5 seconds before the timeout expires.
+            timer = threading.Timer(timeout_seconds - 5,
+                                    lambda: demisto.debug("Timeout reached, dumping the profiler results."))
+            timer.start()
+
+            while True:
+                if signal_event.wait(0.1):  # Check for the event signal
+                    break
+                if not timer.is_alive():  # Check if the timeout occurred
+                    failed_on_timeout = True
+                    break
+
+            profiler.disable()
+            dump_result()
+            demisto.debug("Profiler finished.")
+            return failed_on_timeout
+
+        def function_runner(func, profiler, signal_event,
+                            results, *args, **kwargs):
+            """
+            A wrapper function that runs the targeted profiled function and captures the results in the results list.
+            :param func: The function to be profiled.
+            :param profiler: The Profile object to use for profiling.
+            :param signal_event: The event to signal when profiling is complete.
+            :param results: A shared object to store the results of the profiled function.
+            :param args: The positional arguments to be passed to the profiled function.
+            :param kwargs: The keyword arguments
+            """
+            profiler.enable()
+            demisto.debug("Profiler started.")
+            try:
+                results["function_results"] = func(*args, **kwargs)
+            except Exception as e:
```

Comment:
do we need to explicitly catch exceptions thrown in the function? `try: finally` will be enough to make sure we are freeing the profiler and we dont need to explicitly deal with waht the function we are calling does

Reply:
Nice, didn't know we could do it! 🚀

Reply:
But the issue is that we don't raise the exception we got from the profiled function.